test/async-wait.c

```diff
@@ -0,0 +1,121 @@
+/********************************************************************
+
+   Copyright 2013 Konstantin Olkhovskiy <lupus@oxnull.net>
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+ ********************************************************************/
+
+#include <ev.h>
+#include <check.h>
+#include <evfibers_private/fiber.h>
+
+#include "async-wait.h"
+
+struct fiber_arg {
+	ev_async *async;
+	fbr_id_t *fibers;
+};
+
+static int async_recvd = 0;
+
+static void async_sender(FBR_P_ void *_arg)
+{
+	struct fiber_arg *arg = _arg;
+	ev_async *async = arg->async;
+
+	/* wait briefly to make sure other fiber blocks */
+	fbr_sleep(FBR_A_ 1);
+
+	/* send async event in libev to wake up other fiber */
+	ev_async_send(EV_DEFAULT, async);
```

Comment:
Not particularly relevant to this test case, but using hardcoded EV_DEFAULT might get one into trouble if multiple event loops are in use. It's better to use event loop pointer from the scheduler context.